tests/maint_notifications/test_cluster_maint_notifications_handling.py

```diff
@@ -0,0 +1,343 @@
+from typing import cast
+
+from redis import ConnectionPool, RedisCluster
+from redis.cluster import ClusterNode
+from redis.connection import (
+    BlockingConnectionPool,
+)
+from redis.maint_notifications import MaintNotificationsConfig
+from redis.cache import CacheConfig
+from tests.maint_notifications.proxy_server_helpers import (
+    ProxyInterceptorHelper,
+    RespTranslator,
+)
+
+# Initial cluster node configuration for proxy-based tests
+PROXY_CLUSTER_NODES = [
+    ClusterNode("127.0.0.1", 15379),
+    ClusterNode("127.0.0.1", 15380),
+    ClusterNode("127.0.0.1", 15381),
+]
+
+
+class TestClusterMaintNotificationsConfig:
+    """Test the maint_notifications_config parameter of RedisCluster."""
+
+    # Helper methods
+    def _create_cluster_client(
```

Comment:
Is it possible to re-use existing fixture for client creation? Maybe just add a customisation so new configuration would be passed through